### ISSUE TITLE
Going into SETTINGS then pressing Triangle (Return) generates a black screen.

### ISSUE THREAD

**Issue by Jay-Jay-OPL** [AKuHAK/uLaunchELF#27]
Okay, now testing the new fixes done for Issues #25 and #26  (commit: https://github.com/AKuHAK/uLaunchELF/commit/ff916514596838ed150ae57102f0c970e713fcfb )

And I am using FMCB v1.965 (2018/11/03) and loading BOOT.ELF from MC 1.

I now got a new bug with this wLE version.

**NOTE**: I've compiled this version with the latest changes made to PS2SDK: https://github.com/ps2dev/ps2sdk/commit/fbc3945527d5ddca0cceb5949e95fcca88e1dd64

If I try to go into **SETTINGS** in wLE and then simply press the **TRIANGLE (Return)**, I get a black screen. 

Before, when pressing that button, it should bring me back to wLE main page to select Browser or etc.

It also gives me a black screen after I select OK or Cancel in the SETTINGS page.

**Comment by sp193** [comment]
Please try the one from FMCB v1.965 (2018/11/05):  https://www.sendspace.com/file/xdfrdb
I haven't been able to replicate this and neither should any change cause an inability to transition between screens.

If you don't usually clean your source code, you might want to do ```make clean``` before you close shop, after you update any libraries and after you check out the latest revision.

**Comment by Jay-Jay-OPL** [comment]
Okay, I just tested yours also, and I get the same issue.  

**NOTE**: I am using a uLE Skin with multiple pages.  Maybe that's why?  But that shouldn't be an issue, since when I revert to the last working wLE it works fine.  

The last working wLE version is the one I reported those two issues.  **wLaunchELF_v4.43a (Build-9c18290-2018-11-03)**

And yes, I always run `make clean` whenever I do anything to any local repo -- it's like routine for me on anything.

**Comment by Jay-Jay-OPL** [comment]
**UPDATE**:

I wanted to see if I can verify that the last working wLE doesn't have this bug, and when I tested it with FMCB v1.965 (2018/11/03) - and the bug was still happening -- so I don't know why I didn't spot the bug when I reported those two issues -- weird!

So I uninstalled FMCB and then reinstalled FMCB v1.96**4** (2018/07/22) with wLE: 59a4962 (2018/07/04)

And with that one, I have no issues with wLE and my custom uLE skin.  

Now I plan to now install FMCB v1.965 (2018/11/03) to see what happens?  Maybe the issue could be FMCB v1.965 combined with the latest version of wLE?  Or any new updates to PS2SDK?

**Comment by Jay-Jay-OPL** [comment]
**UPDATE 2:**

Okay, just to bug test wLE 9c18290 with FMCB 1.96**4** -- and the bug returns.  So I didn't even need to install FMCB v1.96**5**.

Which is weird, because at some point I was able to test that wLE version -- hence why I was able to report issues #25 and #26 .  So I am a bit puzzled!?

**Comment by Jay-Jay-OPL** [comment]
Yes, this has been my default custom skin for years -- and no, nothing has changed since then.  But I did test the theory without my skin, and allow uLE to create a new `LAUNCHELF.CNF` file again.  Still the same issue without a custom skin. -- This is why I don't think it's a custom skin issue, since it would fail with the prior wLE versions.

Also keep in mind that even trying to save a new update of the settings causes this black screen in wLE, so it's not only pressing Triangle to return to the main page that has this issue, but I am sure it's the same, it tries to load again the background, but at least the older wLE versions don't seem to have this issue.

Thanks for that tip, I guess FMCB v1.964 (2018/07/22) will fall under that faulty software version.  -- Well, with this issue and the older bugs since June, I'm afraid it's getting quite difficult to find a stable version to rely on.  :-(

**Comment by sp193** [comment]
Just use the November release of FMCB v1.965. There is nothing wrong with it. LaunchELF does not depend on FMCB, so you are free to replace it with whatever you want, although it is good to stay with the latest revision because there isn't a very good version for it.

Where do you keep LaunchELF? I did try to boot it from memory card and my USB disk, but it has no issues.
Have you tried deleting the settings file (LAUNCHELF.CNF), rather than just removing the skin?

**Comment by Jay-Jay-OPL** [comment]
Okay, I keep FMCB installed on MC1 (Slot 2), on MC0 (Slot 1)  -- I keep only game saves.

And I keep wLE's CNF files and the skin images on MC1 (SLOT 2).

I tested this again, and yes, without a skin, I can go into settings and return to the main page without an issue in the aforementioned wLE versions that seem to have a problem.  I'm not sure why I had that issue before, but it seems to be working at least without a skin.

So I think this is why you can't verify the bug, I guess you are not using a skin?

If I use a SKIN, then that is where the issue is.  I just created a fresh `LAUNCHELF.CNF` file, so I am going to compare it with my older CNF files for my skin, to see if I can spot any new changes or whatever.

In the meantime, can you try using a custom skin for uLE, to see if you can confirm the bug?  Here is the skin that I am using: http://www.ps2-home.com/forum/viewtopic.php?f=60&t=2211 --  Just put everything inside your MC1 (Slot 2) memory card.

BTW, I've returned used FMCB v1.96**5** (2018/11/03) -- as you suggested.

**Comment by Jay-Jay-OPL** [comment]
UPDATE:

Okay, doing further testing with the wLE skin.  I am able to find when this bug starts to happen.  It starts with this commit: 2ef0cd4 (Oct. 31, 2018) -- that's when via FTP I only see MC as an option to choose from and no other folders when I use my FTP client.

Prior to this commit, the previous version: 9db3bcc (Oct 20, 2018) and that one via FTP it shows MC and MASS folders.

And a long time ago, during uLE 4.42 it used to show via FTP the following folders: HDD, MASS, MC, and PFS.

So I think if you check again the changes you made to wLE commit: 2ef0cd4 (Oct. 31, 2018) -- you should spot (hopefully?) what went wrong with the SKIN bug and also the FTP not showing all those other folders to choose from.

**Comment by Jay-Jay-OPL** [comment]
**UPDATE**:

I got a headache!  :)

I've been testing this bug with the uLE skin issue for several hours now, I've been testing other uLE versions (some you released and other ones I had to compile going back to older builds).  wLE builds between August 2018 thru Nov 2018.

**The Aftermath**: well it seems that this bug has been around for a while.  And I haven't noticed it until now.  But there is a workaround to avoid it freezing when we press RETURN from going into the Settings. 
 
**WORKAROUND**:
First when you load wLE and with a skin with more than one page, press DPAD left or right to the next page.  What will happen is a brief black screen and then finally you can jump to each page without this black screen.  Once you do that, you can go into the SETTINGS and return anytime.  No problems.  Even with the most latest version of wLE.

So I think this black screen happened when wLE was modded to support TV MODE: **Progressive** scan.  That is when I started to notice that slight black screen when I try to go to the next page on the skin.  It only happens once per session and you no longer get the black screen.

So if you can figure out why the black screen happens right after you load wLE with a multiple skin and you navigate to the next page, you might find the bug in the code??

And the way to get the black screen crash is that you go from loading wLE and then the SETTINGS on the first page, you will get this black screen crash every-time -- but I think only the wLE versions when the progressive scan was added -- I could be wrong?! 

Keep in mind that each CNF file for each SKIN page has to have progressive active if you want to continue using that TV mode, so perhaps that's where the bug is?  And that is what I had each page set to.  Each page had Progressive active.

So now what I did was set Progressive to AUTO per page (to all the custom skin pages).  And now when I launch uLE and press SETTINGS on the first page and then RETURN -- no more bug!  :)  -- I know longer have to conduct that workaround of going to the next page and then return to the first page I want to press SETTINGS for and etc.

So that's the issue!  Damn, It took me a while to figure this out.  I thought I was going crazy, since some older and newer uLE versions had this bug and then it didn't, I couldn't find why it would not fail sometimes and then other times it did, but now I am certain this is it!

**_Sorry for the long reports above, this is the one you need to know..._**

I will later go back to the uLE version that first introduced progressive, to report that one also has the bug, but I've been dealing with this bug for a few hours, and I am hungry... so after a long break, I will give you that final report.

**Comment by sp193** [comment]
Thanks for trying to help.

There was a problem with the setupGS() function not saving the video mode, resulting in the GS being initialized again for the same video mode later on (when you try to leave the settings screen or change the CNF).
I don't know why you'll get "no display" though. But it would be a gsKit thing.

Regarding FTP: it was always like that, by design. The methodology of the uLaunchELF developers was to not load modules, unless they were required.
Perhaps between AKuHAK's WIP builds and now, you may have found the "pfs" device available for access without you needing to access the HDD, due to LaunchELF being made to load the HDD modules at boot to read the network settings. Recently, that was removed because we never standardized a location on the HDD for that and it was actually using FHDB's directory. However, neither FMCB nor FHDB use IPCONFIG.DAT, so I don't see why we should link the two independent pieces of software together for a purpose that never actually existed.
